### ISSUE TITLE
Difficulty with DNA thresholding when metaphase plate is sparse

### ISSUE THREAD

**Issue by gcav1992** [embl-cba/spindle3d#37]
Greetings Spindle3D developers!

First and foremost, just wanted to say how much I appreciate you creating this FIJI plugin. I was taught how to do morphometrics by hand using Imaris, and potentially being able to streamline my workflow is very exciting.

I’ve been trying to implement Spindle3D in a different context than human cells; I am currently working in the _Xenopus laevis_ egg extract system, which recapitulates key aspects of spindle assembly and morphology _in vitro_. However, in comparison to the sample stacks that can be used with Spindle3D to test the plugin, these spindles have much sparser alignment of chromosomes along the metaphase plate. This has lead to repeated issues with Spindle3D DNA thresholding creating DNA masks that are much smaller than the actual metaphase plate, which in turn complicates spindle pole determination and further analysis.

I’ve come to you to inquire about the advanced version of Spindle3D; I feel that this issue could be easily solved if I had control over the thresholding inputs for DNA and Tubulin. I don’t know if this is a pre-existing feature in the advanced version or would require further tweaking, but I foresee this being helpful for any spindle researchers down the road who work with spindles that deviate quite a bit from standard cell culture.

I’ve uploaded 5 different image stacks to a repository at https://github.com/gcav1992/GCM_Spindle3DTest. If you’d like to mess around with these stacks and tinker with parameters, please feel free. They are widefield Z-stacks (of varying quality) taken using SIM (Zeiss Apotome) and are scaled to 16-bit, and I tried to over-expose the DNA slightly to aid in detection.

If you want to discuss anything further or need more information, please let me know – I’d love to be helpful in any way possible.

Regardless, thank you for the time you’ve already put into developing this tool for the spindle community and hope this is a simple request!

All the best,
Gabe Cavin-Meza (Heald Lab @ UC Berkeley)

**Comment by TobiasKletter** [comment]
Hi Gabe! 
Thanks for your question and the images that you provided. 
Yes, the _Xenopus_ extract spindles are really tricky for several reasons, and for Spindle3D mostly a hit-or-miss kind of challenge, largely because of the irregularly shaped chromatin (the morphology of which heavily influences the success of Spindle3D...). 

The strategy for determining the DNA initial threshold first downsamples the voxel sizes such that they barely resolve mitotic chromosome arms (in mammalian cells), and then finds a threshold based on the minimum and maximum pixel value. Lastly, all segmented volumes except for the most central one are excluded. Because the _Xenopus_ chromatin is intrinsically very heterogenous in signal intensity, this step becomes problematic. 

Therefore, looking at your images, I think what you would want is a greatly reduced dynamic range within the chromatin signal, so the originally bright regions stay connected after the DNA initial thresholding step. 
But I do see your point with having the option to set the DNA threshold yourself and it might be worth playing around with this.

However, I would discourage setting the tubulin threshold yourself. The main idea of the plugin is to get rid of user decisions that could be biased or non-reproducible. 

@tischi  I think we never used the option to hard-code the DNA initial threshold, even in the Advanced version, did we? Nevertheless, would it be possible to make the Advanced Version accessible for Gabe with an optionally hard-coded initial DNA threshold? Thanks!

I hope we will be able to help you and your analysis!

**Comment by gcav1992** [comment]
@tischi Thanks for the info! I do agree with @TobiasKletter that altering tubulin threshold is not a great idea, so even just the DNA threshold being adjustable would be fine. No rush and let me know if you need anything from my end!

**Comment by tischi** [comment]
I played around a bit but I think even if we allow to manually provide a DNA threshold it will not work.

Here is how a typical cell looks like for which the code was developed:

<img width="588" alt="image" src="https://user-images.githubusercontent.com/2157566/219675357-8f5a2e4e-1b9d-48b0-8f65-fa6b597a8de5.png">


Here is how one of your cells looks like:

<img width="468" alt="image" src="https://user-images.githubusercontent.com/2157566/219675691-6070401f-ba4c-4772-aa16-e5f3c7f5d6e4.png">

The spindle and DNA are extremely anisotropic in their extent....which I don't think our code can handle....

I could try faking a "longer z-axis" and see if that helps...

**Comment by tischi** [comment]
For such data I wonder whether a 2D analysis would be more appropriate?

@TobiasKletter wasn't there some other issue where we discussed whether one could handle quasi-2D data with our code?

**Comment by TobiasKletter** [comment]
@tischi Yes we discussed it here: #31 and I just re-reading this issue I saw that we even came up with something like a work-around by faking z-dimensions...
 
I think, since the demand for Xenopus spindle analysis is quite high but the 3D analysis so problematic, having a read out of 2D length / width / area for both tubulin and chromatin signals would be helpful.

**Comment by gcav1992** [comment]
@tischi @TobiasKletter I see what you mean and agree that it may be impossible to implement a 3D analysis on these fixed images. I avoided doing a traditional "spin-down" of spindles by centrifugation through a glycerol cushion in the hopes of preserving 3D structure (and thus a more _in vivo_-esque state), but I am beginning to suspect that just the act of placing a slide and laying a coverslip on top of extract causes some of the anisotropy. In addition, I do not have a sense of how much impact the lack of a cell membrane has on overall 3D shape (extract spindles are not bound by any dimensions).

I will be trying to implement Spindle3D in live timelapses that are not placed under a coverslip in the near future - that may help with the anisotropy and be more feasible. For the fixed spindles, however, it may be useful to have a 2D version. However, I'm just concerned about 2D projections and loss of Z-information (i.e. the potential for slightly inaccurate spindle length measurements).

TL;DR - I think, for now, just having a 3D version with adjustable DNA threshold will be helpful for analysis of live imaging spindles that are not fixed and potentially squished (the main goal of the project). It may be useful to have a 2D version for fixed spindles in _Xenopus_ extracts in general, but with the caveat of area measurements instead of volume (and the potential for user bias towards spindles that are most perpendicular to the Z-axis).

**Comment by tischi** [comment]
@gcav1992 

1. OK, I will then anyway upload a 3-D version with manual DNA threshold. I am on holidays now, please ping me again in about two weeks if I did not do it until then!
2. 2-D version is more work and I am not sure how to go about it, it would need more considerations.
3. In terms of how anisotropic it can currently be in the 3-D analysis: we require for the analysis that the shortest axis of the metaphase plate is more or less _parallel_ to the spindle axis. I think, in addition to the threshold, a challenge in your data is that the metaphase plate is so "compressed" along the z-axis that it has more of a cigar shape than a plate shape.  A cigar does have a clear longest axis but it does _not_ have one clear shortest axis and thus our analysis is bound to fail in most of the cases. 
TL;DR for our analysis it really needs to be a metaphase _plate_ ;-)

**Comment by tischi** [comment]
@gcav1992 I uploaded a new version that contains "(Advanced)" versions of the plugins where you can set a DNA threshold factor that will be applied to the automatically found threshold. Does that help you?
